tests/utilities/test_auto_restart.py

```diff
@@ -0,0 +1,588 @@
+    the right next batch on restart.
+    """
+    dataset = range(15)
+    sampler = SequentialSampler(dataset)
+    batch_sampler = BatchSampler(sampler, 3, False)
+    index_batch_sampler = FastForwardSampler(batch_sampler)
+
+    assert isinstance(index_batch_sampler, Iterable)
+    assert has_len(index_batch_sampler)
+
+    index_batch_sampler_iter = iter(index_batch_sampler)
+
+    assert next(index_batch_sampler_iter) == [0, 1, 2]
+    assert next(index_batch_sampler_iter) == [3, 4, 5]
+
+    state_dict = index_batch_sampler.state_dict(2)
+
+    index_batch_sampler = FastForwardSampler(batch_sampler)
+    index_batch_sampler.load_state_dict(state_dict)
+
+    index_batch_sampler_iter = iter(index_batch_sampler)
+    assert next(index_batch_sampler_iter) == [6, 7, 8]
+
+
+def test_fast_forward_on_sequential_sampler():
+    """
+    This test ensures ``FastForwardSampler`` applied to ``SequentialSampler`` correctly retrived
+    the right next batch on restart.
+    """
+    dataset = range(15)
+    sequential_sampler = SequentialSampler(dataset)
+    sampler = FastForwardSampler(sequential_sampler)
+    sampler.setup(3)
+    batch_sampler = BatchSampler(sampler, 3, False)
+
+    batch_sampler_iter = iter(batch_sampler)
+
+    assert next(batch_sampler_iter) == [0, 1, 2]
+    assert next(batch_sampler_iter) == [3, 4, 5]
+
+    state_dict = sampler.state_dict(2)
+    assert state_dict[0]["current_iteration"] == 6
+
+    sampler.load_state_dict(state_dict)
+
+    batch_sampler_iter = iter(batch_sampler)
+    assert next(batch_sampler_iter) == [6, 7, 8]
+
+
+@RunIf(min_torch="1.5.0")
+def test_fast_forward_on_random_sampler():
+    """
+    This test ensures ``FastForwardSampler`` applied to ``RandomSampler`` correctly retrived
+    the right next batch on restart.
+    """
+    seed_everything(42)
+
+    dataset = range(15)
+    random_sampler = RandomSampler(dataset)
+    sampler = FastForwardSampler(random_sampler)
+    sampler.setup(3)
+    batch_sampler = BatchSampler(sampler, 3, False)
+
+    batch_sampler_iter = iter(batch_sampler)
+
+    assert next(batch_sampler_iter) == [14, 9, 1]
+    assert next(batch_sampler_iter) == [7, 11, 3]
+    assert next(batch_sampler_iter) == [12, 8, 2]
+
+    state_dict = sampler.state_dict(3)
+    assert state_dict[0]["current_iteration"] == 9
+    state_dict[0]["current_iteration"] = 6
+
+    seed_everything(42)
+    sampler = FastForwardSampler(random_sampler)
+    sampler.setup(3)
+    batch_sampler = BatchSampler(sampler, 3, False)
+    sampler.load_state_dict(state_dict)
+
+    batch_sampler_iter = iter(batch_sampler)
+    assert next(batch_sampler_iter) == [12, 8, 2]
+    has_raised = False
+    try:
+        for _ in range(5):
+            next(batch_sampler_iter)
+    except StopIteration:
+        has_raised = True
+        assert sampler._current_iteration == 0
+        sampler.load_state_dict(sampler.state_dict(0))
+    assert has_raised
+
+
+class RangeIterativeDataset(IterableDataset):
```

Comment:
Perhaps this should be `RangeIterableDataset`?

Reply:
Done !